Changes to support running rdr test from ibm repo

prsurve opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Enabled disaster recovery workloads to clone from private Git repositories with configurable repository URL/branch and optional authenticated access.
  * DR subscriptions can now reference private-repo credentials via injected Kubernetes Secret references when private-repo login is enabled.
  * Added new Kubernetes Secret templates to support provisioning GitOps private-repository credentials for disaster recovery.
* **Tests**
  * Added an RDR offline-mode safeguard to verify the GitOps private-repo Secret exists for each cluster and create it when missing.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

prsurve committed:
Changes to support running rdr test from ibm repo

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
mask secret when doing git clone

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Add Secret creation mechanism

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
fix getting gh_token

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Add secret check for subscription_app

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Pass secret ref for subscription based app

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Encode secret

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
create key if missing

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
create key if missing

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
scrap few function and implement as per acm

Signed-off-by: prsurve <prsurve@redhat.com>

sidhant-agrawal commented on ocs_ci/ocs/dr/dr_workload.py:
```diff
@@ -46,11 +51,16 @@ class DRWorkload(object):
+        workload_name=None,
+        workload_repo_url=None,
+        workload_repo_branch=None,
+        workload_repo_url_login=None,
```
This is later used as `workload_repo_login`. 
Using `workload_repo_login` here would be consistent

prsurve replied:
fixed

sidhant-agrawal commented on ocs_ci/ocs/dr/dr_workload.py:
```diff
@@ -1080,6 +1148,14 @@ def deploy_workload(self):
+                        ] = constants.PRIVATE_REPO_SUB_SECRET
+                        channel_yaml_data["spec"]["secretRef"][
+                            "namespace"
+                        ] = constants.DEFAULT_NAMESPACE
```
this is not set for BusyBox class above. Could you check if need to add there as well?

prsurve replied:
it's there in busybox class

amr1ta reviewed (dismissed):
looking good, please run a validation job.

prsurve committed:
resolve comments

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
fix tox issue

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
fix tox issue

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
fix tox issue

Signed-off-by: prsurve <prsurve@redhat.com>

shylesh reviewed (dismissed):
lgtm

prsurve committed:
remove ile

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
temp

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
temp

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Replace AUTH github_ibm_odf_qe_ocs_workloads/gh_token with ibm_hci/github_token

Consolidate DR workload GitHub token to a single config location.
All references in dr_workload.py and dr_helpers.py now read from
AUTH[ibm_hci][github_token] instead of
AUTH[github_ibm_odf_qe_ocs_workloads][gh_token].

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Temp change

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Skill bmc verication if required

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Private Repo Fix for cnv

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Private Repo Fix for cnv

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Private Repo Fix for cnv

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Private Repo Fix for cnv

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Private Repo Fix for cnv

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
use providermodecontext if required

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
use providermodecontext if required

Signed-off-by: prsurve <prsurve@redhat.com>

jilju commented on ocs_ci/helpers/helpers.py:
```diff
@@ -7011,18 +7011,26 @@ def find_cephblockpoolradosnamespace(storageclient_uid=None):
+    # StorageConsumer lives on the provider cluster, not the client/hosted cluster.
+    # Use RunWithProviderConfigContextIfAvailable so this works correctly when
+    # called in the context of a hosted (HCP) client cluster.
+    with config.RunWithProviderConfigContextIfAvailable():
```
Kind storageconsumer will automatically use the corresponding provider cluster kubeconfig

jilju commented on ocs_ci/helpers/dr_helpers.py:
```diff
@@ -478,30 +478,36 @@ def check_mirroring_status_ok(
+        # Use RunWithProviderConfigContextIfAvailable so this works correctly when
+        # called in the context of a hosted (HCP) client cluster that has no
+        # cephblockpoolradosnamespaces CRD.
+        with config.RunWithProviderConfigContextIfAvailable():
```
Kind cephblockpoolradosnamespace will automatically use the corresponding provider cluster kubeconfig

prsurve committed:
fix gitleak issue

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
fix gitleak issue

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Resolve Comments

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve committed:
Git leak

Signed-off-by: prsurve <prsurve@redhat.com>

prsurve commented:
opened new one 15807
